pkg/services/sqlstore/migrations/ualert/logzio_ualert_org_migration.go

```diff
@@ -0,0 +1,1242 @@
+package ualert
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/metrics"
+	"github.com/grafana/grafana/pkg/models"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier/channels"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/matttproud/golang_protobuf_extensions/pbutil"
+	"github.com/prometheus/alertmanager/pkg/labels"
+	pb "github.com/prometheus/alertmanager/silence/silencepb"
+	"github.com/prometheus/common/model"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+	"xorm.io/xorm"
+)
+
+// LOGZ.IO GRAFANA CHANGE :: DEV-30705 - Add endpoint to migrate alerts of organization
+const contactPointNameSeparator = "&"
+const contactPointMaxLength = 255
+
+const getDashboardAlertsSql = `
+SELECT id,
+	org_id,
+	dashboard_id,
+	panel_id,
+	org_id,
+	name,
+	message,
+	frequency,
+	%s,
+	state,
+	settings
+FROM
+	alert
+WHERE org_id = ?
+`
+
+//MigrateOrgAlerts migrates old (pre Grafana 8 alerts) to new unified alerts with corresponding notification policies. Based on ualert.migration
+type MigrateOrgAlerts struct {
+	migrator.MigrationBase
+	// session and mg are attached for convenience.
+	sess *xorm.Session
+	mg   *migrator.Migrator
+
+	seenChannelUIDs           map[string]struct{}
+	migratedChannelsPerOrg    map[int64]map[*notificationChannel]struct{}
+	silences                  map[int64][]*pb.MeshSilence
+	portedChannelGroupsPerOrg map[int64]map[string]string // Org -> Channel group key -> receiver name.
+	orgId                     int64
+}
+
+func NewOrgAlertMigration(orgId int64) *MigrateOrgAlerts {
+	return &MigrateOrgAlerts{
+		seenChannelUIDs:           make(map[string]struct{}),
+		migratedChannelsPerOrg:    make(map[int64]map[*notificationChannel]struct{}),
+		portedChannelGroupsPerOrg: make(map[int64]map[string]string),
+		silences:                  make(map[int64][]*pb.MeshSilence),
+		orgId:                     orgId,
+	}
+}
+
+func (m *MigrateOrgAlerts) SQL(dialect migrator.Dialect) string {
+	return "code migration"
+}
+
+//nolint: gocyclo
+func (m *MigrateOrgAlerts) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
```

Comment:
The problem is that, we do not want to make a lot of changes in the code of Grafana because it leads to a lot of conflicts when upgrading to newer version.
The idea with the code is that it will be in the codebase until we migrate all accounts and after that it is going to be removed so that's the reason we want to duplicate some of the code instead of making a lot of changes in the original Grafana code (original code does not have an org id filter for sql queries and we need to refactor it to support both orgId filter and no org id filter so maybe one day we will have grafana instance per customer).

I still think, it's better to have duplicated code for some period of time instead of making a lot of changes which complicates future migrations.

WDYT?

Reply:
Yes, agree